src/test/java/org/openstreetmap/atlas/geography/atlas/change/MemberMergeStrategiesTest.java

```diff
@@ -0,0 +1,333 @@
+         * The merge will fail, since the number of added [2, AREA, areaRole2] conflict.
+         */
+        this.expectedException.expect(CoreException.class);
+        this.expectedException.expectMessage(
+                "diffBasedRelationBeanMerger failed due to ADD/ADD conflict on key: [AREA, 2, areaRole2]: beforeValue absolute count was 0 but addedLeft/Right diff counts conflict [1 vs 2]");
+        MemberMergeStrategies.diffBasedRelationBeanMerger.apply(beforeBean, afterBean1, afterBean2);
+    }
+
+    @Test
+    public void testDiffBasedRelationBeanMergeADDREMOVEConflict()
+    {
+        final RelationBean beforeBean = new RelationBean();
+        beforeBean.addItem(new RelationBeanItem(1L, "pointRole1", ItemType.POINT));
+        beforeBean.addItem(new RelationBeanItem(2L, "pointRole2", ItemType.POINT));
+        beforeBean.addItem(new RelationBeanItem(3L, "pointRole3", ItemType.POINT));
+        beforeBean.addItem(new RelationBeanItem(1L, "lineRole1", ItemType.LINE));
+        beforeBean.addItem(new RelationBeanItem(1L, "areaRole1", ItemType.AREA));
+
+        /*
+         * Add an additional instance of [1, LINE, lineRole1].
+         */
+        final RelationBean afterBean1 = new RelationBean();
+        afterBean1.addItem(new RelationBeanItem(1L, "pointRole1", ItemType.POINT));
+        afterBean1.addItem(new RelationBeanItem(2L, "pointRole2", ItemType.POINT));
+        afterBean1.addItem(new RelationBeanItem(3L, "pointRole3", ItemType.POINT));
+        afterBean1.addItem(new RelationBeanItem(1L, "lineRole1", ItemType.LINE));
+        afterBean1.addItem(new RelationBeanItem(1L, "lineRole1", ItemType.LINE));
+        afterBean1.addItem(new RelationBeanItem(1L, "areaRole1", ItemType.AREA));
+        afterBean1.addItem(new RelationBeanItem(2L, "areaRole2", ItemType.AREA));
+
+        /*
+         * Remove the instance of [1, LINE, lineRole1].
+         */
+        final RelationBean afterBean2 = new RelationBean();
+        afterBean2.addItem(new RelationBeanItem(1L, "pointRole1", ItemType.POINT));
+        afterBean2.addItem(new RelationBeanItem(2L, "pointRole2", ItemType.POINT));
+        afterBean2.addItem(new RelationBeanItem(3L, "pointRole3", ItemType.POINT));
+        afterBean2.addItem(new RelationBeanItem(1L, "areaRole1", ItemType.AREA));
+        afterBean2.addItem(new RelationBeanItem(2L, "areaRole2", ItemType.AREA));
+
+        /*
+         * The merge will fail, since one afterView tries to add an additional [1, LINE, lineRole1]
+         * while the other afterView removes it entirely.
+         */
+        this.expectedException.expect(CoreException.class);
+        this.expectedException.expectMessage(
+                "diffBasedRelationBeanMerger failed due to ADD/REMOVE conflict(s) on key(s): [LINE, 1, lineRole1]");
+        MemberMergeStrategies.diffBasedRelationBeanMerger.apply(beforeBean, afterBean1, afterBean2);
+    }
+
+    @Test
+    public void testDiffBasedRelationBeanMergeREMOVEREMOVEConflict()
```

Comment:
Is REMOVEREMOVE intentional?

Reply:
Ah yes. It's a `REMOVE/REMOVE` conflict, but you can't put a `/` in the name.